### PR TITLE
Add missing functions, Upgrade deprecated `coffee-script` to `coffeescript`, and add Condition support

### PR DESCRIPTION
**Pull request by nysos3**

### CODE REVIEW

File: idx.coffee [hunk 1]
```diff
@@ -3,6 +3,7 @@ Components =
 	Templatize: require('./src/Templatize')
 	Parameter: require('./src/Parameter')
 	Parameterize: require('./src/Parameterize')
+	Condition: require('./src/Parameter')
```
Comment (UnquietCode) [inline idx.coffee]:
```suggestion
	Condition: require('./src/Condition')
```

File: src/Templatize.coffee [hunk 2]
```diff
@@ -47,13 +51,16 @@ descriptionAndArray = (description, array) ->
 		if value instanceof Parameter.__type
 			template.addParameter(value)
 
+		if value instanceof Condition.__type
```
Comment (UnquietCode) [inline src/Templatize.coffee]:
```suggestion
		else if value instanceof Condition.__type
```
NBD since they're distinct types.

**Review by UnquietCode — COMMENTED** [review]
Hey, thanks for the upgrade! These changes look pretty ok to me, except for a couple of lines. I can push a release if needed once it's merged.

File: src/Templatize.coffee (hunk 2 above)
Comment (nysos3) [reply src/Templatize.coffee]: fixed in [c051e58](https://github.com/UnquietCode/Cloud-Temple/pull/2/commits/c051e58ef5d135cb17c4cd6f9f27c1f877a9c6c6)

File: idx.coffee (hunk 1 above)
Comment (nysos3) [reply idx.coffee]: fixed in [c051e58](https://github.com/UnquietCode/Cloud-Temple/pull/2/commits/c051e58ef5d135cb17c4cd6f9f27c1f877a9c6c6)

**Comment by nysos3** [comment]
Pushed a fix with your suggestions. Also added support for Metadata within a template for support of [AWS::CloudFormation::Interface](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cloudformation-interface.html).

**Comment by nysos3** [comment]
@UnquietCode The latest commit adds support for the following Resource attributes:
* CreationPolicy
* UpdatePolicy
* DeletionPolicy
* UpdateReplacePolicy

**Pull request closed without merging**